src/test/scala/firrtlTests/CheckSpec.scala

```diff
@@ -307,6 +307,49 @@ class CheckSpec extends FlatSpec with Matchers {
       }
     }
   }
+
+  s"Duplicate module names" should "throw an exception" in {
+    val input =
+      s"""|circuit bar :
+          |  module bar :
+          |    input i : UInt<8>
+          |    output o : UInt<8>
+          |    o <= i
+          |  module dup :
+          |    input i : UInt<8>
+          |    output o : UInt<8>
+          |    o <= i
+          |  module dup :
+          |    input i : UInt<8>
+          |    output o : UInt<8>
+          |    o <= not(i)
+          |""".stripMargin
+    assertThrows[CheckHighForm.ModuleNameNotUniqueException] {
+      try {
+        checkHighInput(input)
+      } catch {
+        case e: firrtl.passes.PassExceptions => throw e.exceptions.head
+      }
+    }
```

Comment:
Throwing the head of the cause is a clever way to unwrap this.

Reply:
Yeah, it's pretty annoying to make a test cause a single `PassException` in many cases, even when it's possible.

Reply:
Maybe we could write a custom method `assertThrowsOrContainsCause[A <: Throwable]` that will iterate over all throwables until it finds or fails to find type `A`.

Reply:
That's a good idea. I'll add that -- I know I'd use it elsewhere.